src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp

```diff
@@ -780,14 +797,20 @@ namespace xdp {
         msg << "Reserved " << numTraceEvents << " memory trace events for AIE tile (" << col << "," << row << ").";
         xrt_core::message::send(severity_level::debug, "XRT", msg.str());
 
-        if (ret != XAIE_OK) break;
         ret = memoryTrace->setMode(XAIE_TRACE_EVENT_TIME);
-        if (ret != XAIE_OK) break;
-        XAie_Packet  pkt = {0, 1};
+        if (ret != XAIE_OK) 
+          break;
+        XAie_Packet pkt = {0, 1};
         ret = memoryTrace->setPkt(pkt);
-        if (ret != XAIE_OK) break;
+        if (ret != XAIE_OK) 
+          break;
         ret = memoryTrace->start();
-        if (ret != XAIE_OK) break;
+        if (ret != XAIE_OK) 
+          break;
```

Comment:
Reaching at this point I have the feeling to have 2/3 of the code which is
```c++
ret = ...
if (ret != XAIE_OK) 
          break;
```
Ahem... Perhaps there is a pattern here to remove 2/3 of the code? :-)

Reply:
Updated in future PR